Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CertificateChainValidator.java

```diff
@@ -0,0 +1,213 @@
+        return new CertificateChainValidator();
+    }
+
+    /**
+     * ALL root CA are self-sign cert. Verify if cert is self-sign or chain of trust
+     *
+     * @param certToCheck
+     * @return true if self-sign cert or false if chain of trust
+     */
+    public boolean isPartChainOfTrust(X509Certificate certToCheck) {
+        try {
+            // Try to verify certificate signature with its own public key
+            PublicKey key = certToCheck.getPublicKey();
+            certToCheck.verify(key);
+            return false;
+        } catch (SignatureException | InvalidKeyException | CertificateException | NoSuchAlgorithmException
+            | NoSuchProviderException sigEx) {
+            // LOG.warn(sigEx.getLocalizedMessage(),sigEx);
+            return true;
+        }
+
+    }
+
+    /**
+     * Validate chain of trust.
+     *
+     * @param certToCheck
+     * @return
+     */
+    public boolean validateCert(X509Certificate certToCheck) {
+        List<X509Certificate> allCerts = getAllCertInKeyStore(certificateManager.getTrustStore());
+        // check if cert is chain of trust
+        if (isPartChainOfTrust(certToCheck)) {
+            // extract intermediate/root out and compare with trust store
+            // extract intermediate
+            try {
+                // ---------
+                LOG.debug("Cert serial number to verify against our trustStore {}", certToCheck.getSerialNumber()
+                    .toString(16));
+                String leafIssuerString = certToCheck.getIssuerX500Principal().getName();
+                LOG.debug("Issuer for parent {} ", leafIssuerString);
+
+                String authorityKeyIdentifier = getAuthorityKeyIdentify(certToCheck.getExtensionValue(AUTHORITY_KEY_ID));
+                for (X509Certificate cert : allCerts) {
+                    String name = cert.getSubjectX500Principal().getName();
+                    if (leafIssuerString.equalsIgnoreCase(name)) {
+                        String subjectKeyId = getSubjectKeyID(cert.getExtensionValue(SUBJECT_KEY_ID));
+                        if (StringUtils.isNotBlank(subjectKeyId) && StringUtils.isNotBlank(authorityKeyIdentifier)
+                            && subjectKeyId.equalsIgnoreCase(authorityKeyIdentifier)) {
+                            LOG.debug("Found the match with cert serial number{} ", cert.getSerialNumber().toString(16));
+                            // check again if intermediate don't have root
+                            if (isPartChainOfTrust(cert)) {
+                                return validateCert(cert);
+                            } else {
+                                // we should return true since we don't have any parent cert.
+                                return true;
+                            }
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                LOG.error(e.getLocalizedMessage(), e);
```

Comment:
I purposely don't want to throw an exception here.  I just want to give users status so that they can use  to check in CONNECTSamlValidation.  I want to throw their original exception